xonsh/main.py

```diff
@@ -386,6 +372,20 @@ def func_sig_ttin_ttou(n, f):
             code = sys.stdin.read()
             run_code_with_cache(code, shell.execer, glb=shell.ctx, loc=None,
                                 mode='exec')
+        if args.force_interactive:
```

Comment:
yeah, that is how the python interactive option works, you can run a script and then enter the REPL in the same context. It seems like in bash it doesnt make any difference if you use the `-i` flag. Isnt this feature something we want in xonsh?